src/systemctl/systemctl.c

```diff
@@ -7587,6 +7654,38 @@ static int edit(int argc, char *argv[], void *userdata) {
                         continue;
                 }
 
+                /* Trim out the lines between the two markers */
+                r = read_full_file(*tmp, &contents, &size);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to read temporary file \"%s\": %m", *tmp);
+                        goto end;
+                }
+
+                contents_start = strstr(contents, EDIT_MARKER_START);
+                if (contents_start)
+                        contents_start += strlen(EDIT_MARKER_START);
+                else
+                        contents_start = contents;
+
+                contents_end = strstr(contents_start, EDIT_MARKER_END);
+                if (contents_end)
+                        strshorten(contents_start, contents_end - contents_start);
+
+                contents_start = strstrip(contents_start);
+
+                if (isempty(contents_start)) {
+                        log_warning("Editing \"%s\" canceled: temporary file is empty.", *original);
+                        continue;
+                }
+
+                if (strlen(contents_start) != size) {
```

Comment:
hmm?

Reply:
The goal is to avoid re-writing the file if nothing's been changed - i.e. the user has removed both the start and end marker and just left the contents they expect to be in the new unit file.
`contents_start` is checked as the earlier part of the `contents` string isn't modified.
As we're only concerned with trimming the string (between `contents_start` and the NUL terminator), a `strlen` comparison should be sufficient.
Hopefully my reasoning is sound.
If you're happy to leave it in I'll add a clarification comment for others reading the code.